UnitTests/CommonTestUtils/ConfigureJoinableTaskFactoryAttribute.cs

```diff
@@ -1,26 +1,41 @@
 ﻿using System;
 using System.Collections;
+using System.Diagnostics;
 using System.Runtime.CompilerServices;
 using System.Runtime.ExceptionServices;
 using System.Threading;
 using System.Windows.Forms;
+using GitCommands;
 using GitUI;
 using Microsoft.VisualStudio.Threading;
 using NUnit.Framework;
 using NUnit.Framework.Interfaces;
+using ResourceManager;
 
 namespace CommonTestUtils
 {
     [AttributeUsage(AttributeTargets.Assembly)]
     public sealed class ConfigureJoinableTaskFactoryAttribute : Attribute, ITestAction
     {
+        private readonly bool _checkForUpdatesOriginalValue = AppSettings.CheckForUpdates;
+
         private DenyExecutionSynchronizationContext _denyExecutionSynchronizationContext;
         private ExceptionDispatchInfo _threadException;
 
         public ActionTargets Targets => ActionTargets.Test;
 
+        public static bool IgnoreExceptions { get; set; }
+
         public void BeforeTest(ITest test)
         {
+            AppSettings.CheckForUpdates = false;
+
+            // Ignore instances from previous failed tests.
+            GitExtensionsFormBase.InstanceRegistry.Clear();
+
+            Assert.IsNull(ThreadHelper.JoinableTaskContext, "Tests with joinable tasks must not be run in parallel!");
```

Comment:
I think we can enforce this with https://github.com/nunit/docs/wiki/NonParallelizable-Attribute

Reply:
I also believe tests in the same category are run serially, so we could also leverage categories https://github.com/nunit/docs/wiki/Category-Attribute

Reply:
`[Apartment(ApartmentState.STA)]` seems to supress parallel execution -- as long as the tests are contained in the same test assembly. This is or might become an issue with the partitioned UI tests.

I added this check in order to inform a future user if their configuration does not suit.